src/Elastic.Apm/Model/Span.cs

```diff
@@ -135,14 +141,25 @@ internal Span StartSpanInternal(string name, string type, string subType = null,
 
 			if (!string.IsNullOrEmpty(action)) retVal.Action = action;
 
-			_logger.Debug()?.Log("Starting {SpanDetails}", retVal.ToString());
+			_logger.Trace()?.Log("Starting {SpanDetails}", retVal.ToString());
 			return retVal;
 		}
 
 		public void End()
 		{
-			_logger.Debug()?.Log("Ending {SpanDetails}", ToString());
-			if (!Duration.HasValue) Duration = (DateTimeOffset.UtcNow - _start).TotalMilliseconds;
+			if (Duration.HasValue)
+			{
+				_logger.Debug()?.Log("End() called on already ended {Span}. Start time: {Time} (as timestamp: {Timestamp}), Duration: {Duration}ms",
+					this, TimeUtils.FormatTimestampForLog(Timestamp), Timestamp, Duration);
+			}
+			else
+			{
+				var endTimestamp = TimeUtils.TimestampNow();
+				Duration = TimeUtils.DurationBetweenTimestamps(Timestamp, endTimestamp);
+				_logger.Trace()?.Log("Ended {Span}. Start time: {Time} (as timestamp: {Timestamp}), End time: {Time} (as timestamp: {Timestamp}), Duration: {Duration}ms",
```

Comment:
Any reason this is `Trace` and above is `Debug`?

Reply:
That was under assumption that `Duration` can be set only by `End` (which i think should be the case) and thus calling `End` more than once is very unusual and we might want to know about it.